src/admin-dhis2/components/AdminDhis2DatasetForm/AdminDhis2DatasetForm.jsx

```diff
@@ -1,98 +1,245 @@
-import React, { useMemo } from 'react';
+import React, { useMemo, useState, useEffect } from 'react';
 
-import { Form } from 'react-final-form';
-import arrayMutators from 'final-form-arrays';
-import { FieldArray } from 'react-final-form-arrays';
+import { toast } from 'react-toastify';
 
 import getService from '../../../react-components/utils/angular-utils';
-import SelectField from '../../../react-components/form-fields/select-field';
-import AddButton from '../../../react-components/buttons/add-button';
+import { SearchSelect } from '../../../requisition-order-create/search-select'
 
-function AdminDhis2DatasetForm({
-    onSubmit, onCancel, 
-    title, initialFormValue, 
-    mode, refetch
-}) {
+function AdminDhis2DatasetForm({ onSubmit, onCancel, refetch, serverId }) {
 
-  // TODO - here fetch from backend datasets that the DHIS2 user has access 
-  const datasetOptions = [];  
+    const cronExpressionOptions = [
+        { name: 'MONTHLY', value: 'MONTHLY' },
+        { name: 'WEEKLY', value: 'WEEKLY' },
+        { name: 'DAILY', value: 'DAILY' }
+    ];
 
-  const validate = values => {
-    const errors = { items: [] };
+    const weeklyOptions = 7;
+    const dayOptions = 31;
+    const hourOptions = 23;
+    const minuteOptions = 59;
 
-    return errors;
-};
+    const [datasetOptions, setDatasetOptions] = useState([]);
 
-  const serverService = useMemo(
-      () => {
-          return getService('adminDhis2');
-      },
-      []
-  );
+    const [dhisDataset, setDhisDataset] = useState('')
+    const [selectedDataset, setSelectedDataset] = useState("");
+    const [selectedCronExpression, setSelectedCronExpression] = useState("");
+    const [timeOffset, setTimeOffset] = useState(0);
+    const [day, setDay] = useState(1);
+    const [hour, setHour] = useState(0);
+    const [minutes, setMinutes] = useState(0);
+
+    const serverService = useMemo(
+        () => {
+            return getService('adminDhis2');
+        },
+        []
+    );
+
+    const fetchDatasets = () => {
+        if (serverId) {
+            serverService?.getServerDatasetsList(serverId)
+                .then((fetchedDatasets) => {
+                    const {content} = fetchedDatasets;
+
+                    const datasets = content.map((dataset) => ({
+                        name: dataset.displayName,
+                        value: dataset.displayName,
+                        id: dataset.id
+                    }));
+                    setDatasetOptions(datasets);
+                });
+        }
+    }
+
+    useEffect(() => fetchDatasets(), [serverId]);
+
+    const selectNumber = (value, countFrom = 0) => {
+
+        const options = [];
+        let n = countFrom;
+
+        for(n; n <= value ; n++) {
+            options.push({ name: `${n}`, value: n });
+        }
+
+        return options;
+    }
+
+    const selectDate = () => (
+        <>
+            {selectedCronExpression === 'MONTHLY' &&
```

Comment:
I think here we can take advantage of having const CRON_EXPRESSION_OPTIONS and assign here CRON_EXPRESSION_OPTIONS[0]['value'] instead of having hardcoded 'Monthly'

Reply:
updated